### ISSUE TITLE
[NBug] Diff tab crash

### ISSUE THREAD

**Issue by GufNZ** [gitextensions/gitextensions#11168]
### Environment

- Git Extensions 4.1.0.16698
- Build 1fe52a137acef1fbd2b653dea263b44ad5844177
- Git 2.42.0.windows.1
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.21
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

Exit code: 128
Command: C:\Program Files\Git\cmd\git.exe
Arguments: -c color.ui=never -c diff.submodule=short -c diff.noprefix=false -c diff.mnemonicprefix=false -c diff.ignoreSubmodules=none -c core.safecrlf=false diff --find-renames --find-copies "2e364d3392c0106b223f65f388da495e5a1128ff" "32e73751e0433235c143ee941e2365b1cd0e122f" -- "modules/jai_parser"
Working directory: C:\Jails\

```
GitExtUtils.ExternalOperationException
 ---> System.Exception
   --- End of inner exception stack trace ---
   at async Task<ExecutionResult> GitCommands.ExecutableExtensions.ExecuteAsync(IExecutable executable, ArgumentString arguments, Action<StreamWriter> writeInput, Encoding outputEncoding, CommandCache cache, bool stripAnsiEscapeCodes, bool throwOnErrorExit, CancellationToken cancellationToken) in C:/projects/gitextensions/GitCommands/Git/ExecutableExtensions.cs:line 323
   at async Task<Patch> GitCommands.GitModule.GetSingleDiffAsync(ObjectId firstId, ObjectId secondId, string fileName, string oldFileName, string extraDiffArguments, Encoding encoding, bool cacheResult, bool isTracked) in C:/projects/gitextensions/GitCommands/Git/GitModule.cs:line 2399
   at async Task<GitSubmoduleStatus> GitCommands.Git.SubmoduleHelpers.GetCurrentSubmoduleChangesAsync(GitModule module, string fileName, string oldFileName, ObjectId firstId, ObjectId secondId) in C:/projects/gitextensions/GitCommands/Git/SubmoduleHelpers.cs:line 13
   at async void GitCommands.GitModule.GetSubmoduleStatus(IReadOnlyList<GitItemStatus> status, ObjectId firstId, ObjectId secondId)+(?) => { } in C:/projects/gitextensions/GitCommands/Git/GitModule.cs:line 2735
   at async Task GitUI.GitUIExtensions.ViewChangesAsync(FileViewer fileViewer, FileStatusItem item, CancellationToken cancellationToken, int? line, string defaultText, Action openWithDiffTool, string additionalCommandInfo)+GetSelectedPatchAsync(?) in C:/projects/gitextensions/GitUI/GitUIExtensions.cs:line 140
   at async Task GitUI.GitUIExtensions.ViewChangesAsync(FileViewer fileViewer, FileStatusItem item, CancellationToken cancellationToken, int? line, string defaultText, Action openWithDiffTool, string additionalCommandInfo) in C:/projects/gitextensions/GitUI/GitUIExtensions.cs:line 91
   at async Task GitUI.CommandsDialogs.RevisionDiffControl.ShowSelectedFileDiffAsync(bool ensureNoSwitchToFilter, int? line) in C:/projects/gitextensions/GitUI/CommandsDialogs/RevisionDiffControl.cs:line 605
   at async void GitUI.CommandsDialogs.RevisionDiffControl.ShowSelectedFile(bool ensureNoSwitchToFilter, int? line)+(?) => { } in C:/projects/gitextensions/GitUI/CommandsDialogs/RevisionDiffControl.cs:line 617
   at async void GitUI.TaskManager.FileAndForget(Task task, Func<Exception, bool> fileOnlyIf)+(?) => { } in C:/projects/gitextensions/GitExtUtils/GitUI/TaskManager.cs:line 29
```

Tried to switch to the diff tab for the first time and got this, after a dialog showing the command that failed.
Args: -c color.ui=never -c diff.submodule=short -c diff.noprefix=false -c diff.mnemonicprefix=false -c diff.ignoreSubmodules=none -c core.safecrlf=false diff --find-renames --find-copies "2e364d3392c0106b223f65f388da495e5a1128ff" "32e73751e0433235c143ee941e2365b1cd0e12 2f" -- "modules/jai_parser"
ExitCode: 128

### Steps to reproduce

Installed gitForWindows (also had an old cygwin install - deleting it's git didn't change anything)
Installed gitExtensions
cloned a repo
switched to the Diff tab.

### Did this work in previous version of GitExtensions?

I've never had this issue with gitExt before at work, tho this is the first time installing on my home machine.

**Comment by GufNZ** [comment]
Works on another repo.  Maybe something odd about that one..: https://github.com/SogoCZE/Jails

**Comment by gerhardol** [comment]
It is Git that fails to diff the file, GE just displays the error message
Related to #11164

The Git error message should probably be shown in the diff viewer

**Comment by mstv** [comment]
Works for me:

```
D:\Build\Jails>git -c color.ui=never -c diff.submodule=short -c diff.noprefix=false -c diff.mnemonicprefix=false -c diff.ignoreSubmodules=none -c core.safecrlf=false diff --find-renames --find-copies "2e364d3392c0106b223f65f388da495e5a1128ff" "32e73751e0433235c143ee941e2365b1cd0e122f" -- "modules/jai_parser"
diff --git a/modules/jai_parser b/modules/jai_parser
index 0186eb2..d989d97 160000
--- a/modules/jai_parser
+++ b/modules/jai_parser
@@ -1 +1 @@
-Subproject commit 0186eb29046680fc2c8f55740d53698b1e1adad3
+Subproject commit d989d973e915b5acbc7f4c3100a2e30709b61e4a

D:\Build\Jails>echo %ERRORLEVEL%
0
```

![image](https://github.com/gitextensions/gitextensions/assets/36601201/4a775849-0920-4dd9-80d4-f81046367aad)

So it might be related to some settings of GE or git
You could run a recent portable build for comparison, e.g. from https://ci.appveyor.com/api/buildjobs/9n8mr82cq3fgcvgh/artifacts/artifacts%2FRelease%2Fpublish%2FGitExtensions-Portable-4.2.0.17242-7e00ab283.zip.
`git config -l`

**Comment by GufNZ** [comment]
```
diff.astextplain.textconv=astextplain
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
http.sslbackend=openssl
http.sslcainfo=C:/Program Files/Git/mingw64/etc/ssl/certs/ca-bundle.crt
core.autocrlf=true
core.fscache=true
core.symlinks=true
core.editor="C:\\Program Files\\Microsoft VS Code\\bin\\code" --wait
pull.rebase=false
credential.helper=manager
credential.https://dev.azure.com.usehttppath=true
init.defaultbranch=master
user.email=j.chris.findlay@gmail.com
user.name=J.Chris Findlay
core.repositoryformatversion=0
core.filemode=false
core.bare=false
core.logallrefupdates=true
core.ignorecase=true
submodule.active=.
remote.origin.url=https://github.com/GufNZ/jai.git
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
branch.main.remote=origin
branch.main.merge=refs/heads/main
```

**Comment by mstv** [comment]
The following are unset in my setup. I have not digged into the git docs. But the first and the last look like good candidates for experiments.

```
core.repositoryformatversion=0
core.filemode=false
core.bare=false
core.logallrefupdates=true
core.ignorecase=true
submodule.active=.
```

**Comment by GufNZ** [comment]
Seems some things git wants aren't in the windows path - will fix that and try again.

**Comment by GufNZ** [comment]
...can't find extDiff anywhere - not sure what it uses instead in cygwin..

**Comment by pmiossec** [comment]
> ...can't find extDiff anywhere

I don't know if it makes reference to https://github.com/ForNeVeR/ExtDiff
If it's the case, maybe you have something defined in `.gitattributes`.

> - not sure what it uses instead in cygwin..

yeah, they are mysteries like that...

**Comment by gerhardol** [comment]
Closed in #11175 
Error message is now in the diff viewer
